src/platform/packages/shared/kbn-esql-ast/src/pretty_print/basic_pretty_printer.ts

```diff
@@ -162,6 +180,62 @@ export class BasicPrettyPrinter {
     return formatted;
   }
 
+  protected decorateWithSubstitutedParameters<T extends ESQLAstBaseItem>(
+    node: T,
+    formatted: string
+  ): string {
+    if (isColumn(node)) {
+      const argParam = node.args.find((arg) => isParamLiteral(arg));
+      if (argParam) {
+        formatted = this.replaceLiteralWithParameter(argParam, formatted);
+      }
+    } else if (isLiteral(node) && isParamLiteral(node)) {
+      formatted = this.replaceLiteralWithParameter(node, formatted);
+    }
```

Comment:
There is also a case where parameter is a function name:

```
FROM a | STATS ?func(1, 2, 3)
```

Could you please handle that, too.